### PR TITLE
Redesign congrats page

### PR DESCRIPTION
**Pull request by juanmanzojr**
Redesigns the /congrats page. The page now has two different views: Student View and Teacher/signed out view.
Students see different courses in the "Beyond an Hour of Code" section than the Teacher/Signed out users.

Teachers/Signed out users will also have one button in the "Discover More" section while students will have two.

Teachers/Signed out users will have a Teach/Professional Learning section.

### Certificate and Input section
Screenshot 2023-10-24 at 1 42 15 PM

### Student view of "Beyond an Hour of Code" section
Screenshot 2023-10-24 at 1 50 39 PM

### Teacher/Signed out view of "Beyond an Hour of Code" section
Screenshot 2023-10-24 at 2 11 01 PM

### Teach/Professional Learning section for Teachers/Signed out users
Screenshot 2023-10-24 at 1 54 53 PM

## Links

- jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?selectedIssue=ACQ-1036)
- figma: [here](https://www.figma.com/file/19X0B5VYIVBZDd3Z8lU1eq/HoC-2023?type=design&node-id=1809-1538&mode=design&t=a980rgsADlaLxObf-0)

## Testing story
Local

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/certificates/Congrats.jsx [hunk 1]
```diff
@@ -72,97 +59,250 @@ export default function Congrats(props) {
...
     isHocTutorial,
-    nextCourseScriptName,
-    nextCourseTitle,
-    nextCourseDesc,
   } = props;
 
-  const isEnglish = language === 'en';
-  const tutorialType = getTutorialType(tutorial);
+  // const isEnglish = language === 'en';
+  // const tutorialType = getTutorialType(tutorial);
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/Congrats.jsx]: Can we remove these if we don't need them anymore?

File: apps/src/templates/certificates/Congrats.jsx [hunk 2]
```diff
@@ -72,97 +59,250 @@ export default function Congrats(props) {
...
+                {i18n.learningForAgesRange({
+                  youngestAge: '5',
+                  oldestAge: '11',
+                })}
+              </a>
+              <a
+                className={`${style.linkButton} ${style.catalogButton}`}
+                href={'https://code.org/student/middle-high'}
+              >
+                {i18n.learningForAgesPlus({age: '11'})}
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/Congrats.jsx:254]: Thanks for doing these dynamic strings!!

File: apps/src/templates/certificates/Congrats.jsx [hunk 3]
```diff
@@ -72,97 +59,250 @@ export default function Congrats(props) {
...
+          {curriculaData.map((item, index) => (
+            <div
+              className={`${style.actionBlock} ${style.actionBlockOneCol} ${style.flexSpaceBetween}`}
+              key={index}
+            >
+              <div className={style.contentWrapper}>
+                <p className={style.overline}>{item.grade}</p>
+                <Heading3>{item.title}</Heading3>
+                <img src={item.image} alt="" />
+                <p>{item.description}</p>
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/Congrats.jsx]: Similar to using the `Heading3` tags, can you also use `BodyOneText` (or `BodyTwoText`, `BodyThreeText`, etc. if one of those looks more right visually) for `<p>` tags in here

File: apps/src/templates/certificates/SocialShare.jsx [hunk 4]
```diff
@@ -85,4 +82,10 @@ const styles = {
     color: color.white,
     minWidth: 40,
   },
+  printButton: {
+    backgroundColor: 'transparent',
+    borderColor: color.black,
+    borderWidht: '1px',
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/SocialShare.jsx]: Typo in `borderWidht`, does fixing that change the visual output too much?

File: apps/src/templates/certificates/certificate_batch.module.scss [hunk 5]
```diff
@@ -267,3 +267,37 @@ a.linkButton {
...
+  max-width: $width-md;
+  margin-left: auto;
+  margin-right: auto;
+}
+
+.divider {
+  border-color: $lightest_gray;
+  border-width: 1px 0 0 0;
+  border-style: solid;
+  margin: 20px 0 20px 0;
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/certificate_batch.module.scss]: Nit: you can specify top/bottom, left/right if they're the same, so this line can be `margin: 20px 0`

**Review by TurnerRiley — APPROVED** [review]
A few nits, but overall looking amazing!

File: apps/test/unit/templates/certificates/CongratsTest.js [hunk 6]
```diff
@@ -23,45 +19,42 @@ describe('Congrats', () => {
...
-      );
-      expect(wrapper.find(PetitionCallToAction).exists()).to.be.true;
-      expect(wrapper.find(PetitionCallToAction).props().tutorial).to.not.be
-        .undefined;
-    });
+  it('renders two learning for ages buttons, for students', () => {
+    const wrapper = shallow(<Congrats {...defaultProps} userType="student" />);
+
+    wrapper.find('a[href="https://code.org/student/elementary"]');
+    wrapper.find('a[href="https://code.org/student/middle-high"]');
```
Comment (TurnerRiley) [inline apps/test/unit/templates/certificates/CongratsTest.js]: Can you wrap these in `expect(wrapper ... ]').exists()).to.be.true;` statements like the other tests please

File: apps/test/unit/templates/certificates/CongratsTest.js [hunk 7]
```diff
@@ -23,45 +19,42 @@ describe('Congrats', () => {
...
+  it('renders curriculum catalog button, for teachers', () => {
+    const wrapper = shallow(<Congrats {...defaultProps} userType="teacher" />);
 
-    it(`renders a GraduateToNextLevel for user type ${userType} for CSF course`, () => {
-      const wrapper = shallow(
-        <Congrats {...defaultProps} userType={userType} isHocTutorial={false} />
-      );
-      expect(wrapper.find(GraduateToNextLevel).exists()).to.be.true;
-    });
+    wrapper.find('a[href="/catalog"]');
```
Comment (TurnerRiley) [inline apps/test/unit/templates/certificates/CongratsTest.js]: Can you wrap this in `expect(wrapper ... ]').exists()).to.be.true;` statements like the other tests please

File: apps/src/templates/certificates/Certificate.jsx [hunk 8]
```diff
@@ -131,53 +138,73 @@ function Certificate(props) {
...
+                >
+                  {i18n.submit()}
+                </button>
+              </div>
+            </div>
+          )}
+          {tutorial && personalized && (
+            <div>
+              <Heading2>
+                <div id="uitest-thanks">{i18n.congratsCertificateThanks()}</div>
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/Certificate.jsx:191]: Is this id made for ui tests being used anywhere?

File: apps/src/templates/certificates/Certificate.jsx [hunk 9]
```diff
@@ -131,53 +138,73 @@ function Certificate(props) {
...
-        )}
-        <h2>{i18n.congratsCertificateShare()}</h2>
-        <SocialShare
-          facebook={facebook}
-          twitter={twitter}
-          print={print}
-          under13={under13}
-        />
+      <div className={style.certificateContainer}>
+        <div id="uitest-certificate" className={certificateStyle}>
```
Comment (TurnerRiley) [inline apps/src/templates/certificates/Certificate.jsx:154]: Same for this ui test ID

File: apps/src/templates/certificates/Certificate.jsx (hunk 8 above)
Comment (juanmanzojr) [reply apps/src/templates/certificates/Certificate.jsx:191]: Yes its being used in `hour_of_code_finish.feature`. The drone build was failing as the Heading component doesnt allow for ids as a prop.

File: apps/src/templates/certificates/Certificate.jsx (hunk 9 above)
Comment (juanmanzojr) [reply apps/src/templates/certificates/Certificate.jsx:154]: Same goes here. Id is Being used within `hour_of_code_finish.feature`.

**Pull request merged**